packages/devtools-source-map/src/source-map.js

```diff
@@ -306,6 +306,45 @@ async function getOriginalSourceText(originalSource: Source) {
   };
 }
 
+async function getFileGeneratedRange(originalSource: Source) {
+  assert(isOriginalId(originalSource.id), "Source is not an original source");
+  const sourceText = originalSource.text;
+  if (!sourceText) {
+    return null;
+  }
+
+  const start = await getGeneratedLocation(
+    { sourceId: originalSource.id, line: 1, column: 0 },
+    originalSource
+  );
+
+  const lines = sourceText.split("\n");
+
+  const endRanges = await getGeneratedRanges(
```

Comment:
i spoke to @loganfsmyth and we can drop this approach in favor of something much simpler...

for the end location we should get the generatedPosition for the end line/column with a SourceMapConsumer.LEAST_UPPER_BOUND bias. This should just be one call..